### ISSUE TITLE
Support resize for amp-pinterest

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#10318]
Currently `amp-pinterest` does not support resizing meaning if the initial width/height provided is wrong, component will get cut off . 

pinterest API provides the real width/height of the image and we already have that data, so we should `attemptChangeHeight` change if provided width/height are different than that pinterest reports ( of course `attemptChangeHeight` is not gauranteed and only works in item is not inside viewport)

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it. @wassgha Please add an appropriate category.

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it. @wassgha Please add an appropriate category.

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it. @wassgha Please add an appropriate category.

**Comment by westonruter** [comment]
+1 from me on this. I've just implemented AMP-compatibility for the Pinterest block in the Jetpack plugin for WordPress. In order to avoid the Pin getting cut off I had to go through a whole bunch of tedious work to fetch the Pin info from Pinterest's REST API and factor in hard-coded size values from AMP's pinterest stylesheet: 

https://github.com/Automattic/jetpack/blob/690a58590ade44141d7b795381264c0351b51e8c/extensions/blocks/pinterest/pinterest.php#L102-L150

While such embeds should have accurate initial dimensions, it is not practical to expect users to provide precise dimensions for the embed, especially as they can change over time. So `amp-pinterest` should support auto-resizing in the same way as `amp-twitter` or `amp-gist`. True this will negatively impact CLS, but this needs to be an area where AMP is pragmatic.
